app/models/piece.rb

```diff
@@ -1,3 +1,83 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def current_pieces_coords
+    
+    self.game.pieces.map { |piece| [piece.x_coord, piece.y_coord] }
+    
+  end
+
+  def is_obstructed?(x, y)
+    if x_coord == x && y_coord != y
+      is_vertically_obstructed?(x, y)
+    elsif x_coord != x && y_coord == y
+      is_horizontally_obstructed?(x, y)
+    elsif (x_coord - x).abs == (y_coord - y).abs && x_coord != x && y_coord != y
+      is_diagonally_obstructed?(x, y)
+    else
+      return 'invalid input square'   # Maybe raise? used return for my tests
+    end
+  end
+
+  def is_vertically_obstructed?(x, y) 
+    y_range = []                    #!!! need to refactor without using array, also diagonally #Brandon
```

Comment:
@Brandongoodman615 another idea for creating ranges:

Using a ternary operator (https://www.thoughtco.com/ternary-or-conditional-operator-2907827) instead of if/else could make it a bit more concise and readable..

```
y_range = y_coord < y ? ((y_coord+1)..(y-1)).to_a : ((y+1)..(y_coord-1)).to_a
```
This is because ` (y_coord+1..y-1).each { |n| y_range << n }  `is essentially converting a range into an array..